### PR TITLE
feat(BBD-833): Connect CVS Past Purchase work to live API

### PR DESCRIPTION
**Pull request by zfcb**
groupby/flux-capacitor#139
groupby/storefront-core#150
groupby/storefront-products#28
groupby/storefront-sort#18
groupby/storefront-structure#41
groupby/storefront-navigation#32
groupby/storefront-recommendations#8
https://github.com/groupby/storefront-presets/pull/52
https://github.com/groupby/storefront-page-size/pull/17
https://github.com/groupby/storefront-paging/pull/19

### CODE REVIEW

File: src/products/index.ts [hunk 1]
```diff
@@ -17,27 +18,52 @@ export type Transformer = (product: Product) => { data: object, variants: object
...
+    switch (this.props.storeSection) {
+      case StoreSections.PAST_PURCHASES:
+        this.set({ products: this.mapProducts(products) });
+        break;
+      case StoreSections.DEFAULT:
+        this.set({
+          products: this.select(Selectors.productsWithPastPurchase, this.config.recommendations.idField)
+            .map(this.productTransformer)
+        });
+    }
```
Comment (treeforever) [inline src/products/index.ts]: need default to fall back? Maybe spit out an error message?

File: src/products/index.ts [hunk 2]
```diff
@@ -17,27 +18,52 @@ export type Transformer = (product: Product) => { data: object, variants: object
...
+  updateProducts = (products: any = []) => {
+    switch (this.props.storeSection) {
+      case StoreSections.PAST_PURCHASES:
+        this.set({ products: this.mapProducts(products) });
+        break;
+      case StoreSections.DEFAULT:
+        this.set({
+          products: this.select(Selectors.productsWithPastPurchase, this.config.recommendations.idField)
+            .map(this.productTransformer)
+        });
```
Comment (treeforever) [inline src/products/index.ts]: need "break"

File: src/products/index.ts [hunk 3]
```diff
@@ -17,27 +18,52 @@ export type Transformer = (product: Product) => { data: object, variants: object
...
 
   init() {
-    this.flux.on(Events.PRODUCTS_UPDATED, this.updateProducts);
+    switch (this.props.storeSection) {
+      case StoreSections.PAST_PURCHASES:
+        this.flux.on(Events.PAST_PURCHASE_PRODUCTS_UPDATED, this.updateProducts);
+        break;
+      case StoreSections.DEFAULT:
+        this.flux.on(Events.PRODUCTS_UPDATED, this.updateProducts);
+    }
```
Comment (treeforever) [inline src/products/index.ts]: if you are calling the same update function, why do you need a switch? To add cases in the future?

File: src/products/index.ts [hunk 4]
```diff
@@ -17,27 +18,52 @@ export type Transformer = (product: Product) => { data: object, variants: object
...
+    storeSection: StoreSections.DEFAULT
+  };
+
   state: Products.State = {
     products: []
   };
   productTransformer: Transformer = ({ data, meta }: Product) =>
     ({ ...ProductTransformer.transformer(this.structure)(data), meta })
 
   init() {
```
Comment (treeforever) [inline src/products/index.ts]: You might want to move init() above prodctTransformer as all your other methods are underneath

File: src/products/index.ts (hunk 3 above)
Comment (Filip-F) [reply src/products/index.ts]: They're called on different events though.

File: src/products/index.ts (hunk 4 above)
Comment (Filip-F) [reply src/products/index.ts]: Done.

File: src/products/index.ts [hunk 5]
```diff
@@ -17,26 +18,46 @@ export type Transformer = (product: Product) => { data: object, variants: object
...
   };
+
+  init() {
+    switch (this.props.storeSection) {
+      case StoreSections.PAST_PURCHASES:
+        this.flux.on(Events.PAST_PURCHASE_PRODUCTS_UPDATED, this.updateProducts);
+        break;
+      case StoreSections.SEARCH:
+        this.flux.on(Events.PRODUCTS_UPDATED, this.updateProducts);
+      break;
```
Comment (treeforever) [inline src/products/index.ts]: 2 more spaces :)

File: src/products/index.ts [hunk 6]
```diff
@@ -17,26 +18,46 @@ export type Transformer = (product: Product) => { data: object, variants: object
...
+    }
+  }
+
   productTransformer: Transformer = ({ data, meta }: Product) =>
     ({ ...ProductTransformer.transformer(this.structure)(data), meta })
 
-  init() {
-    this.flux.on(Events.PRODUCTS_UPDATED, this.updateProducts);
+  updateProducts = (products: any = []) => {
+    switch (this.props.storeSection) {
```
Comment (treeforever) [inline src/products/index.ts]: why not make separate update functions for different section of the store? Switch cases are not saving any code here

File: src/products/index.ts [hunk 7]
```diff
@@ -17,26 +18,41 @@ export type Transformer = (product: Product) => { data: object, variants: object
...
   updateProducts = () =>
     this.set({
       products: this.select(Selectors.productsWithPastPurchase, this.config.recommendations.idField)
         .map(this.productTransformer)
     })
+
+  updatePastPurchaseProducts = (products: any = []) =>
+    this.set({ products: this.mapProducts(products) })
+
+  mapProducts = (products: Store.ProductWithMetadata[]) => {
```
Comment (effervescentia) [inline src/products/index.ts]: why do we need to split this into two functions?

**Pull request merged**